scripts/irods/upgrade_configuration.py

```diff
@@ -346,15 +346,24 @@ def update_base(base, updates):
         new_server_config["tcp_keepalive_time_in_seconds"] = service_account_environment.get("irods_tcp_keepalive_time_in_seconds", None)
 
     # Update database plugin config (if present) to new format
+    # TODO: We should probably check the existence of these items before
+    # checking the len
     if len(new_server_config['plugin_configuration']['database']) != 0:
         # Only focus on part we care about
         database_config_options = new_server_config['plugin_configuration']['database']
 
         # Assume we have a previously valid config (i.e. one database config property)
-        db_name = database_config_options.keys()[0]
+        # MAYBE: If there are other configurations, we should delete them
+        # or just do something (e.g. print a warning?)
+        db_configurations = list(database_config_options.keys())
+        db_name = db_configurations.pop(0)
```

Comment:
In the case there are multiple database configurations, do we:

A) Leave the configurations there and continue
B) (A) but also print or log a warning
C) Remove all other configurations, defaulting to the first
D) Error out
E) Something else?

Currently, I'm defaulting to the first configuration encountered and leaving the rest untouched, if that helps.

Reply:
To answer my own question, I think I'll go with option B unless suggested otherwise. The schema should be executed and find something wrong at some point. The additional logs/warnings might help too. Arbitrarily choosing the first option is a bit funky, though...

---

Perhaps being stricter with Option D is better. Then we don't pick a config, since there's only one there now.

Reply:
B) seems best.  I don't know how we handle multiple... so we should go with 'one' and be noisy about the others.

Choosing something other than the first seems more funky.

Reply:
I think I'm pretty on the edge for either option B or option D. I'm a bit of a fan of having the person upgrading iRODS choose the specific configuration they want (by erroring out until they fix it).

I'll implement option B tomorrow if no one else has an opinion on it. :+1:

Reply:
I agree that option (B) sounds the most appropriate. Log it and continue as if everything worked. The server will catch any issues on startup.

What happens to the other entries though? Are they discarded or forwarded somehow?

Reply:
The other entries would be forwarded.

Reply:
Done.